probeinterface/probe.py

```diff
@@ -0,0 +1,259 @@
+        else:
+            raise ValueError()
+        self.set_shape_vertices(vertices)
+    
+    def set_device_channel_indices(self, channel_indices):
+        """
+        Set the channel indices on device side.
+        
+        If some channel are not connected or not recorded then channel can be "-1"
+        """
+        if channel_indices.size != self.get_electrode_count():
+            valueError('channel_indices have not the same size as electrode')
+        self.device_channel_indices = channel_indices
+        if self._probe_bunch is not None:
+            self._probe_bunch.check_global_device_wiring()
+    
+    def copy(self):
+        """
+        Copy to another Probe instance.
+        
+        Note: device_channel_indices is not copied.
+        """
+        other = Probe()
+        other.set_electrodes(
+                    positions=self.electrode_positions.copy(),
+                    plane_axes=self.electrode_plane_axes.copy(),
+                    shapes=self.electrode_shapes.copy(),
+                    shape_params=self.electrode_shape_params.copy())
+        if self.probe_shape_vertices is not None:
+            other.set_shape_vertices(self.probe_shape_vertices.copy())
+        # channel_indices are not copied
+        return other
+
+    def to_3d(self, plane='xz'):
+        """
+        Transform 2d probe to 3d probe.
+        
+        Note: device_channel_indices is not copied.
+        
+        Parameters
+        ----------
+        plane: 'xy', 'yz' ', xz'
+        """
+        assert self.ndim ==2
+        
+        probe3d = Probe(ndim=3, si_units=self.si_units)
+        
+        # electrodes
+        positions = _2d_to_3d(self.electrode_positions, plane)
+        plane0 = _2d_to_3d(self.electrode_plane_axes[:, 0, :], plane)
+        plane1 = _2d_to_3d(self.electrode_plane_axes[:, 1, :], plane)
+        plane_axes = np.concatenate([plane0[:, np.newaxis, :], plane1[:, np.newaxis, :]], axis=1)
+        probe3d.set_electrodes(
+                    positions=positions,
+                    plane_axes=plane_axes,
+                    shapes=self.electrode_shapes.copy(),
+                    shape_params=self.electrode_shape_params.copy())
+
+        # shape
+        if self.probe_shape_vertices is not None:
+            vertices3d = _2d_to_3d(self.probe_shape_vertices, plane)
+            probe3d.set_shape_vertices(vertices3d)
+        
+        if self.device_channel_indices is not None:
+            probe3d.device_channel_indices = self.device_channel_indices
+        
+        return probe3d
+    
+    def rotate(self, axis, theta, origin):
+        """
+        Rorate the probe the specified axis
+
+        Parameters
+        ----------
+        axis
+        
+        theta
+        
+        origin
+        """
+        if self.ndim == 2:
+            raise NotImplementedError
+        elif self.ndim == 3:
+            raise NotImplementedError
+        
+        
+    
+    def move(self, direction):
```

Comment:
Direction is a bit misleading as it is more of a translation vector. I would either put distance and direction or I would just put something like translation_vector.

I would also call this function translate (more inline with the rotate function).

Reply:
OK for translation_vector
move was taken from the MEAutility package of alessio
@alejoe91 : "move" or "translate"